index.bs

```diff
@@ -1370,77 +896,768 @@ What configuration is the device in after it resets?
+        write data received in response to this transfer.
+    1. Let |bytesTransferred| be the number of bytes written to |buffer|.
+    1. Let |result| be a new {{USBInTransferResult}}.
+    1. If data was received from <a>this</a> create a new {{ArrayBuffer}} containing
+        the first |bytesTransferred| bytes of |buffer| and set
+        <code>|result|.{{USBInTransferResult/data}}</code> to a new {{DataView}}
+        constructed over it.
+    1. If <a>this</a> responded by stalling the
+        <a>default control pipe</a> set
+        <code>|result|.{{USBInTransferResult/status}}</code> to {{"stall"}}.
+    1. If <a>this</a> responded with more than |length| bytes of data set
+        <code>|result|.{{USBInTransferResult/status}}</code> to {{"babble"}} and
+        otherwise set it to {{"ok"}}.
+    1. If the transfer fails for any other reason <a>reject</a> |promise| with a
+        {{NetworkError}} and abort these steps.
+    1. <a>Resolve</a> |promise| with |result|.
+</div>
+
+<div algorithm="USBDevice.controlTransferOut(setup, data)">
+    The <dfn for=USBDevice method>controlTransferOut(|setup|, |data|)</dfn> method,
+    when invoked, must return a new {{Promise}} |promise| and run the following
+    steps <a>in parallel</a>:
+    1. <a>Check if the device is configured</a> with <a>this</a> and |promise|
+        and abort these steps if promise is rejected.
+    1. <a>Check the validity of the control transfer parameters</a> with <a>this</a>
+        and abort these steps if |promise| is rejected.
+    1. Issue a <a>control transfer</a> with the <a>setup packet</a> populated by
+        |setup| and the data transfer direction in <code>bmRequestType</code> set
+        to "host to device" and <code>wLength</code> set to
+        <code>|data|.length</code>. Transmit |data| in the <a>data stage</a> of
+        the transfer.
+    1. Let |result| be a new {{USBOutTransferResult}}.
+    1. If the device responds by stalling the
+        <a>default control pipe</a> set
+        <code>|result|.{{USBOutTransferResult/status}}</code> to {{"stall"}}.
+    1. If the device acknowledges the transfer set
+        <code>|result|.{{USBOutTransferResult/status}}</code> to {{"ok"}} and
+        <code>|result|.{{USBOutTransferResult/bytesWritten}}</code> to
+        <code>|data|.length</code>.
+    1. If the transfer fails for any other reason <a>reject</a> |promise| with a
+        {{NetworkError}} and abort these steps.
+    1. <a>Resolve</a> |promise| with |result|.
+</div>
+
+<div algorithm="USBDevice.clearHalt(direction, endpointNumber)">
+    The <dfn for=USBDevice method>clearHalt(|direction|, |endpointNumber|)</dfn> method,
+    when invoked, MUST return a new {{Promise}} |promise| and run the following steps
+    <a>in parallel</a>:
+    1. <a>Check if the device is configured</a> with <a>this</a> and |promise|
+        and abort these steps if promise is rejected.
```

Comment:
```suggestion
        and abort these steps if |promise| is rejected.
```

Reply:
Updated in https://github.com/WICG/webusb/pull/212/commits/5aaf696938ed1dc073f8cc8da8c809e9e5019f41.